reqwest-tracing/src/reqwest_otel_span_macro.rs

```diff
@@ -122,23 +123,26 @@ macro_rules! reqwest_otel_span {
             let url = $request.url();
             let scheme = url.scheme();
             let host = url.host_str().unwrap_or("");
-            let host_port = url.port().unwrap_or(0) as i64;
+            let host_port = url.port_or_known_default().unwrap_or(0) as i64;
             let otel_name = $name.to_string();
+            let header_default = &::http::HeaderValue::from_static("");
+            let user_agent = format!("{:?}", $request.headers().get("user_agent").unwrap_or(header_default)).replace('"', "");
 
             macro_rules! request_span {
                 ($lvl:expr) => {
                     $crate::reqwest_otel_span_macro::private::span!(
                         $lvl,
                         "HTTP request",
-                        http.method = %method,
-                        http.scheme = %scheme,
-                        http.host = %host,
-                        net.host.port = %host_port,
+                        http.request.method = %method,
+                        url.scheme = %scheme,
+                        server.address = %host,
+                        server.port = %host_port,
+                        url.full = %url,
```

Comment:
I believe I made a mistake here where adding the full URL at least wasn't something that the `DefaultSpanBackend` did and instead only the `SpanBackendWithUrl`, which I believe is still using the old key by accident.

Reply:
Thank you for pointing this out 🙌 
https://github.com/TrueLayer/reqwest-middleware/pull/141/commits/58a8e4ca4b220bca096bf1d12f4dc05967b1740b